data/gen/main.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/ipld/go-ipld-prime/schema"
+	gengo "github.com/ipld/go-ipld-prime/schema/gen/go"
+)
+
+func main() {
+	ts := schema.TypeSystem{}
+	ts.Init()
+	adjCfg := &gengo.AdjunctCfg{}
+
+	pkgName := "data"
+
+	ts.Accumulate(schema.SpawnString("String"))
+	ts.Accumulate(schema.SpawnInt("Int"))
+	ts.Accumulate(schema.SpawnBytes("Bytes"))
+
+	ts.Accumulate(schema.SpawnList("BlockSizes", "Int", false))
+
+	/*
+		type UnixTime struct {
+			seconds Int
+			fractionalNanoseconds Int
+		}
+	*/
+	ts.Accumulate(schema.SpawnStruct("UnixTime",
+		[]schema.StructField{
+			schema.SpawnStructField("Seconds", "Int", false, false),
+			schema.SpawnStructField("FractionalNanoseconds", "Int", true, false),
+		},
+		schema.SpawnStructRepresentationMap(nil),
+	))
+
+	/*
+		type UnixFSData struct {
+			dataType Int
+			data optional Bytes
+			filesize optional Int;
+			blocksizes [Int]
+
+			hashType optional Int
+			fanout optional Int
+			mode optional Int
+			mtime optional UnixTime
+		} representation map
+	*/
+
+	ts.Accumulate(schema.SpawnStruct("UnixFSData",
+		[]schema.StructField{
+			schema.SpawnStructField("DataType", "Int", false, false),
+			schema.SpawnStructField("Data", "Bytes", true, false),
+			schema.SpawnStructField("FileSize", "Int", true, false),
+			schema.SpawnStructField("BlockSizes", "BlockSizes", false, false),
+			schema.SpawnStructField("HashType", "Int", true, false),
+			schema.SpawnStructField("Fanout", "Int", true, false),
+			schema.SpawnStructField("Mode", "Int", true, false),
+			schema.SpawnStructField("Mtime", "UnixTime", true, false),
+		},
+		schema.SpawnStructRepresentationMap(nil),
+	))
```

Comment:
note to @warpfork we really should figure out how to generate this intermediate code-gen format from ipld schemas :)